kani-driver/src/concrete_playback.rs

```diff
@@ -18,42 +21,64 @@ use std::process::Command;
 
 impl KaniSession {
     /// The main driver for generating concrete playback unit tests and adding them to source code.
-    pub fn gen_and_add_concrete_playback(&self, output_filename: &Path, harness: &HarnessMetadata) {
+    pub fn gen_and_add_concrete_playback(
+        &self,
+        harness: &HarnessMetadata,
+        verification_output: &VerificationOutput,
+    ) -> Result<()> {
         if self.args.concrete_playback.is_none() {
-            return;
+            return Ok(());
         }
 
-        assert!(
+        ensure!(
             self.args.output_format != crate::args::OutputFormat::Old,
             "The Kani argument `--output-format old` is not supported with the concrete playback feature."
         );
 
-        let concrete_vals = parser::extract_concrete_vals(output_filename).expect(
-            "Something went wrong when trying to get concrete values from the CBMC output file",
-        );
-        let concrete_playback = format_unit_test(&harness.mangled_name, &concrete_vals);
+        if verification_output.status == VerificationStatus::Success {
+            if !self.args.quiet {
+                println!(
+                    "INFO: The concrete playback feature does not generate unit tests when verification succeeds."
```

Comment:
Since we have removed the `--harness [h]` requirement, I think this info here might be too verbose.

Reply:
I'm not sure I understand why `--harness` impacts the verbosity here.

Regardless of whether that's required or not, I see how this could be a bit too verbose. I'm unclear what the correct behavior should be here: 1) print INFO, 2) quietly skip generating a unit test, 3) print ERROR and panic. I think 3) is too much. I assume that if the user inputted this flag, they expect a test case to be generated. So if its not, 2) is too passive, and we should at least tell them with 1) (the current approach). Do you have thoughts on this?

Reply:
Well, if I run `cargo kani` in a project where all proofs succeed (the common case if Kani is integrated in CI), this message would be printed once per proof, so I agree it's a little too verbose. What about printing if `--verbose`? Another option is to change `cargo kani` to print a message (only one, at the end) if the user passed the concrete playback flag but all proofs passed.

Reply:
Ah, that makes sense. It would be much less verbose when you have a bunch of successful harnesses and only one failing harness.

I updated the code to add a `inform_if_no_failed` function. This informs the user at the very end if there are no failed harnesses.